### DISCUSSION TITLE
Async: Discuss

### DISCUSSION THREAD

**Discussion by e-dant** [e-dant/watcher#15] [General]
Hello.

Do you know things?

Are those things about asynchronous design?

This is where you'd write those things. 

This is a discussion thread for the best practices and patterns in asynchronous design.

I don't want this project to end up in, or contribute to, callback hell. 

But I also don't want to wait for senders and receivers (when's that happening, anyway?)

**Comment by e-dant** [comment]
We could boost::asio-ify this project, but that would break the whole "dependency free" pitch.

Is it worth it?

**Comment by klemens-morgenstern** [comment]
I won't use it if it's not asio. You'll need to add so much code to make this work, that you'll be asio-lite yourself. It's pointless and a maintenance nightmare that is not worth the "dependency free" pitch. You'll be the dependency.

Just support asio standalone as well, and your min dependency is a header-only library.
